### PR TITLE
WooCommerce: Handle legacy JS use of ES6

### LINKED ISSUE #38296: pnpm run build failing at current trunk
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

```
@woocommerce/components:turbo:build: cache miss, executing 6985fccda580978b
command (/home/helga/woocommerce/plugins/woocommerce/client/legacy) pnpm run turbo:build exited (1)

 Tasks:    2 successful, 20 total
Cached:    0 cached, 20 total
  Time:    6.775s 

 ERROR  run failed: command  exited (1)
 ELIFECYCLE  Command failed with exit code 1.
```

### Expected behavior

build scripts finish compiling and the plugin has all assets

### Actual behavior

the WC plugin unusable with the following fatal error:

```
Fatal error: Uncaught Exception: Could not find asset registry for wp-admin-scripts in /user/woocommerce/plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php:128 Stack trace: #0 /user/woocommerce/plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php(412): Automattic\WooCommerce\Internal\Admin\WCAdminAssets::get_script_asset_filename('wp-admin-script...', 'variable-produc...') #1 /user/woocommerce/plugins/woocommerce/includes/admin/class-wc-admin-pointers.php(76): Automattic\WooCommerce\Internal\Admin\WCAdminAssets::register_script('wp-admin-script...', 'variable-produc...', true) #2 /user/woocommerce/plugins/woocommerce/includes/admin/class-wc-admin-pointers.php(41): WC_Admin_Pointers->create_variable_product_tutorial() #3 /app/wp-includes/class-wp-hook.php(308): WC_Admin_Pointers->setup_pointers_for_screen('post-new.php') #4 /app/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters(NULL, Array) #5 /app/wp-includes/plugin.php(517): WP_Hook->do_action(Array) #6 /app/wp-admin/admin-header.php(118): do_action('admin_enqueue_s...', 'post-new.php') #7 /app/wp-admin/edit-form-advanced.php(425): require_once('/app/wp-admin/a...') #8 /app/wp-admin/post-new.php(75): require('/app/wp-admin/e...') #9 {main} thrown in /user/woocommerce/plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php on line 128 
```

### Steps to reproduce

pull latest trunk
delete `node_modules` and pnpm lock file
run `pnpm install` and then `pnpm run build`

### WordPress Environment

pnpm v8.5.1
PHP 8.1.7 (in command line)

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by psealock**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/38296

This PR uses `grunt-contrib-uglify-es` instead of `grunt-contrib-uglify` to make use of ES6 features. Note that it does not handle optional chaining and null coalescence, so I manually changed instances of those in the code.

The best way to handle this would be to introduce Babel. I didn't do this to avoid configuring and adding complexity to the Grunt task. Considering this is for handling legacy JS that will eventually be phased out, I'm ok with it. But, I'm open to hearing an opposing view.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. `rm -rf node_modules` 
2. `rm pnpm-lock.yaml`
3. `pnpm install`
4. `pnpm run build`

### CODE REVIEW

**Comment by lsinger** [comment]
Running `rm -rf node_modules && rm pnpm-lock.yaml && pnpm install` followed by `pnpm run build` gives me an error (and a changed `pnpm-lock.yaml`):

```
...
woocommerce/client/legacy:turbo:build: Running "copy:js" (copy) task
woocommerce/client/legacy:turbo:build: Created 20 directories, copied 71 files
woocommerce/client/legacy:turbo:build: 
woocommerce/client/legacy:turbo:build: Running "uglify:js_assets" (uglify) task
@woocommerce/e2e-core-tests:turbo:build: >> Building package: @woocommerce/e2e-core-tests
package-release:turbo:build: src/commands/prepare/index.ts(4,10): error TS2305: Module '"@oclif/core"' has no exported member 'CliUx'.
package-release:turbo:build: src/commands/publish/index.ts(4,10): error TS2305: Module '"@oclif/core"' has no exported member 'CliUx'.
package-release:turbo:build:  ELIFECYCLE  Command failed with exit code 2.
package-release:turbo:build: ERROR: command finished with error: command (/Users/leif/Code/woocommerce/tools/package-release) pnpm run turbo:build exited (1)
@woocommerce/internal-js-tests:turbo:build:  ELIFECYCLE  Command failed.
@woocommerce/e2e-core-tests:turbo:build:  ELIFECYCLE  Command failed.
woocommerce/client/legacy:turbo:build:  ELIFECYCLE  Command failed.
@woocommerce/e2e-utils:turbo:build:  ELIFECYCLE  Command failed.
@woocommerce/notices:turbo:build:  ELIFECYCLE  Command failed.
@woocommerce/e2e-core-tests:turbo:build:  ELIFECYCLE  Command failed.
monorepo-merge:turbo:build:  ELIFECYCLE  Command failed.
@woocommerce/api:turbo:build:  ELIFECYCLE  Command failed.
@woocommerce/e2e-utils:turbo:build:  ELIFECYCLE  Command failed.
@woocommerce/csv-export:turbo:build: cache miss, executing 87fae05be872432d
@woocommerce/date:turbo:build: cache miss, executing a6628bd65fb11910
@woocommerce/navigation:turbo:build: cache miss, executing de355b6eefb8977a
@woocommerce/number:turbo:build: cache miss, executing f3117f81cd1cc352
@woocommerce/explat:turbo:build: cache miss, executing 900075777608a4ca
@woocommerce/e2e-environment:turbo:build: cache miss, executing f373bdb1282bbe42
command (/Users/leif/Code/woocommerce/tools/package-release) pnpm run turbo:build exited (1)

 Tasks:    3 successful, 17 total
Cached:    0 cached, 17 total
  Time:    3.064s 

 ERROR  run failed: command  exited (1)
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by samueljseay** [comment]
Hmm sorry to drive-by but these test instructions probably are not completely correct, please bear in mind that if you delete the lock then create a new one with a (possibly) different version of pnpm you might (most likely will) get a lock file with different resolved dependencies (esp because we have version ranges with `^` which change latest version quite frequently).

It should work with the existing pnpm lock file. 

I'm not sure what you're trying to clear out as well. if you rm node_modules it often won't do anything because pnpm caches those dependencies.

File: plugins/woocommerce/client/legacy/js/admin/wc-enhanced-select.js [hunk 1]
```diff
@@ -312,7 +312,7 @@ jQuery( function( $ ) {
 					var select2_args = $.extend( {
 						allowClear        : $( this ).data( 'allow_clear' ) ? true : false,
 						placeholder       : $( this ).data( 'placeholder' ),
-						minimumInputLength: $( this ).data( 'minimum_input_length' ) ?? '3',
```
Comment (ObliviousHarmony) [inline plugins/woocommerce/client/legacy/js/admin/wc-enhanced-select.js:315]: With the change to `uglify-es` do we need to make these changes?

Comment (psealock) [reply plugins/woocommerce/client/legacy/js/admin/wc-enhanced-select.js:315]: yes, that syntax is beyond the es 6 standard and requires other additional tooling to compile.

**Comment by ObliviousHarmony** [comment]
Just as a note @psealock, all of the checks in this PR are failing on the uglification step!

**Comment by psealock** [comment]
@ObliviousHarmony there was another null coalescence occurrence addressed in  https://github.com/woocommerce/woocommerce/pull/38342/commits/e56ba6711831bca596e011642bc67c5fb5fc7a84

**Review by ObliviousHarmony — APPROVED** [review]

**Pull request merged**